ReactWindows/ReactNative/Views/Image/ReactImageManager.cs

```diff
@@ -107,41 +112,48 @@ public void SetResizeMode(Border view, string resizeMode)
         /// Set the source URI of the image.
         /// </summary>
         /// <param name="view">The image view instance.</param>
-        /// <param name="source">The source URI.</param>
+        /// <param name="sources">The source URI.</param>
         [ReactProp("src")]
-        public void SetSource(Border view, string source)
+        public void SetSource(Border view, JArray sources)
         {
-            var imageBrush = (ImageBrush)view.Background;
+            // There is no image source
+            if (sources.Count == 0)
+            {
+                return;
+            }
 
-            view.GetReactContext()
-                .GetNativeModule<UIManagerModule>()
-                .EventDispatcher
-                .DispatchEvent(
-                    new ReactImageLoadEvent(
-                        view.GetTag(),
-                        ReactImageLoadEvent.OnLoadStart));
+            // Optimize for the case where we have just one uri, case in which we don't need the sizes
+            if (sources.Count == 1)
+            {
+                var uri = ((JObject)sources[0]).Value<string>("uri");
+                SetUriFromSingleSource(view, uri);
+            }
+            else
+            {
+                var viewSources = default(Dictionary<string, double>);
+                if (_imageSources.TryGetValue(view.GetTag(), out viewSources))
+                {
+                    viewSources.Clear();
```

Comment:
Can you put back your logic with TryGetValue where you clear existing list if available?  I think we decided it would be better than reallocating the list each time.

---
In reply to: [76172342](https://github.com/ReactWindows/react-native-windows/pull/625#discussion_r76172342) [](ancestors = 76172342)